### PR TITLE
Setting up associations completed

### PR DESCRIPTION
**Pull request by ometman**
I have completed the creation of associations in branch 'associations'

**Created a class Classroom with the following:**
- @label instance variable, initialized in the constructor.
- Setter and getter for @label using attr_accessor.

**Created has-many/belongs-to relationship between Classroom and Student**
- Create the has-many side (a classroom has many students).
- Create the belongs-to side (a student belongs to a classroom).
- When adding a student to a classroom it also sets the classroom for the student.
- When setting the classroom for a student it also adds it to the classrooms' students.

**Created a class Book with the following:**
- @title and @author instance variables, initialized in the constructor.
- Setters and getters for instance variables using attr_accessor.

**Create a class Rental with the following:**
- @date instance variable, in the constructor.
- Setter and getter for @date using attr_accessor.

**Created a many-to-many (has-many-through) relationship between Person and Book using the intermediate class Rental.**
- Created the has-many side of Book and Rental (a book has many rentals).
- Created the belongs-to side of Rental and Book (a rental belongs to a book).
- Created the has-many side of Person and Rental (a person has many rentals).
- Created the belongs-to side of Rental and Person (a rental belongs to a person).
- Modified the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: rental.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class Rental
+  attr_accessor :date, :book, :person
+
+  def initialize(date, book, person)
+    @date = date
+    @book = book
+    @person = person
+  end
```
Comment (MahmoudBakr23) [inline rental.rb]: - [ ] Please make sure to assign the rentals to the book and person instances for the record to be saved in the Rental model class when it's invoked by the other two. 🙏 

    @book = book
    book.rentals << self

    @person = person
    person.rentals << self

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @ometman 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Well done on creating the correct associations between the classes so far
- Good job with the custom methods
- Descriptive PR and README
- Correct Linters


However, there is always room for improvement.

> I'd like to remind you that your long hours of work are appreciated. However, the code review process is for improving your code, not criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

N/A

Cheers, and Happy coding!👏👨‍💻💻

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: rental.rb (hunk 1 above)
Comment (ometman) [reply rental.rb]: Yeah sure. 😄

Comment (rloterh) [reply rental.rb]: issue resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Ometman,  

Your project is complete! 🏁 There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 🎯 
🎲 Good job on implementing recommended changes!

## Optional suggestions 🧰 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

**You can also consider:**
Let's improve the look of the ReadMe by removing the underlined:

image

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ometman** [comment]
Thanks

**Pull request merged**